eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/ReplicatedCassandra.scala

```diff
@@ -52,60 +56,104 @@ object ReplicatedCassandra {
     statements: Statements[F],
     expiryService: ExpiryService[F],
   ): ReplicatedJournal[F] = {
+    applyWithTempMetrics(segmentSizeDefault, segmentNrsOf, statements, expiryService, None, None)
+  }
+
+  private def applyWithTempMetrics[F[_]: Sync: Parallel: Fail](
+    segmentSizeDefault: SegmentSize,
+    segmentNrsOf: SegmentNrsOf[F],
+    statements: Statements[F],
+    expiryService: ExpiryService[F],
+    metrics: Option[ReplicatedJournal.Metrics[F]],
+    log: Option[Log[F]],
+  ): ReplicatedJournal[F] = {
 
     new Main with ReplicatedJournal[F] {
 
-      def topics = {
+      val unit: F[Unit] = Applicative[F].unit
+
+      def topics: F[SortedSet[Topic]] = {
         statements
           .selectTopics2()
           .parProduct(statements.selectTopics())
-          .map { case (a, b) => a ++ b }
+          .flatMap {
+            case (a, b) =>
+              val meterAndLog = Applicative[F].whenA(b.diff(a).nonEmpty) {
+                val diff = b.diff(a)
```

Comment:
Minor thing: you're doing `b.diff(a)` twice, might be reasonable to move
```
val diff = b.diff(a)
```
out of `.whenA(...)` scope and use in in the condition and body.